src/main/java/com/flightstats/hub/webhook/InternalWebhookClient.java

```diff
@@ -0,0 +1,110 @@
+package com.flightstats.hub.webhook;
+
+import com.flightstats.hub.app.HubHost;
+import com.flightstats.hub.cluster.CuratorCluster;
+import com.flightstats.hub.rest.RestClient;
+import com.flightstats.hub.util.HubUtils;
+import com.google.common.annotations.VisibleForTesting;
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.ClientResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.*;
+
+import static java.util.stream.Collectors.toList;
+
+public class InternalWebhookClient {
+    private final static Logger logger = LoggerFactory.getLogger(InternalWebhookClient.class);
+
+    private final Client client;
+    private final CuratorCluster hubCluster;
+
+    @Inject
+    public InternalWebhookClient(@Named("HubCuratorCluster") CuratorCluster hubCluster) {
+        this.hubCluster = hubCluster;
+        this.client = RestClient.createClient(5, 15, true, true);
+    }
+
+    @VisibleForTesting
+    InternalWebhookClient(Client client, CuratorCluster hubCluster) {
+        this.hubCluster = hubCluster;
+        this.client = client;
+    }
+
+    List<String> remove(String name, Collection<String> servers) {
+        return servers.stream()
+                .filter(server -> remove(name, server))
+                .collect(toList());
+    }
+
+    boolean remove(String name, String server) {
+        return put(server + "/internal/webhook/delete/" + name);
+    }
+
+    Optional<String> runOnServerWithFewestWebhooks(String name) {
+        return runOnOneServer(name, getOrderedServers());
+    }
+
+    Optional<String> runOnOneServer(String name, Collection<String> servers) {
+        return servers.stream()
+                .filter(server -> put(server + "/internal/webhook/run/" + name))
+                .findFirst();
```

Comment:
Doesn't this send a `PUT` to every server instead of just one?

Reply:
`findFirst` short-circuits on the first element that meets the requirement. I added a test to make sure that's true, though if it's not intuitive, I'm happy to switch back to the loop + break.

Reply:
Interesting, I didn't know that. Cool. I looked at the unit test for it and it looks good.